controllers/awsadapterconfig_controller.go

```diff
@@ -58,9 +58,31 @@ func (r *AWSAdapterConfigReconciler) Reconcile(ctx context.Context, req ctrl.Req
 	l := log.FromContext(ctx)
 
 	objOld := &securityv1alpha1.AWSAdapterConfig{}
-	err := r.Get(ctx, types.NamespacedName{Name: req.Name, Namespace: req.Namespace}, objOld)
+	err := r.Get(ctx, req.NamespacedName, objOld)
 	if err != nil {
-		l.Error(err, "error occurred while retrieving awsadapterconfig")
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	awsacfgFinalizer := "security.nirmata.io/finalizer"
+	if objOld.DeletionTimestamp.IsZero() {
```

Comment:
Why do you need a finalizer? Isn't objOld.Deletionstamp.IsZero() check not enough to print a warning that the CR is being deleted, and return from there?